airflow/models/pool.py

```diff
@@ -62,6 +71,45 @@ def get_default_pool(session=None):
         """
         return Pool.get_pool(Pool.DEFAULT_POOL_NAME, session=session)
 
+    @staticmethod
+    @provide_session
+    def slots_stats(session=None) -> Dict[str, PoolStats]:
+        from airflow.models.taskinstance import TaskInstance  # Avoid circular import
+
+        pools: Dict[str, PoolStats] = {}
+
+        pool_rows: Iterable[Tuple[str, int]] = session.query(Pool.pool, Pool.slots).all()
+        for (pool_name, total_slots) in pool_rows:
+            pools[pool_name] = PoolStats({
+                "total": total_slots,
+                State.RUNNING: 0,
+                State.QUEUED: 0,
+            })
+
+        state_count = (
+            session.query(TaskInstance.pool, TaskInstance.state, func.count())
+            .filter(TaskInstance.state.in_(list(EXECUTION_STATES)))
+            .group_by(TaskInstance.pool, TaskInstance.state)
+        ).all()
```

Comment:
Oh, I find it, could we just use
```suggestion
        state_count = (
            session.query(TaskInstance.pool, func.count())
            .filter(TaskInstance.state.in_(list(EXECUTION_STATES)))
            .group_by(TaskInstance.pool)
        ).all()
```

here? and then we could not use `# calculate queued and used metrics`, we just need to update the `pools` dict by `state_count`? And maybe change the variable name for more easy to know?

Reply:
i have removed the if state check in that for loop, it's not needed anymore with explicit state filter in the query.

with regards to your proposed change: if we remove group by state from the query, how do we get queued slot count for a particular pool?

Reply:
> with regards to your proposed change: if we remove group by state from the query, how do we get queued slot count for a particular pool?

Ohh, my bad, I think we still need group by state here